src/puppetlabs/comidi.clj

```diff
@@ -21,17 +21,24 @@
 (def Zipper
   (schema/pred ks/zipper?))
 
-(def http-methods
-  #{:any :get :post :put :delete :head :options})
-
 (def RequestMethod
   (schema/enum :any :get :post :put :delete :head :options))
 
+; Derived from bidi-schema PatternSegment
 (def RegexPatternSegment
-  [(schema/one Pattern "regex") (schema/one schema/Keyword "variable")])
+  (schema/pair schema/Regex "qual" schema/Keyword "id"))
+
+; Derived from bidi-schema by flattening a vector of Patterns
+(def RouteInfoPathSegment
+  (schema/either bidi-schema/Path
+                 bidi-schema/PatternSegment
+                 schema/Bool))
```

Comment:
Tangent, but interesting: just saw a PR go through prismatic's `plumbing` library where they got rid of uses of `either` because apparently it is officially deprecated now:

https://github.com/Prismatic/plumbing/pull/106/files